packages/neuron-ui/src/components/AmendPendingTransactionDialog/hooks.ts

```diff
@@ -0,0 +1,236 @@
+      } finally {
+        setSending(false)
+      }
+    } catch {
+      // ignore
+    }
+  }, [walletID, tx, setShowConfirmedAlert, setPwdError, password, generatedTx, setSending])
+
+  return {
+    fee,
+    price,
+    setPrice,
+    generatedTx,
+    setGeneratedTx,
+    transaction,
+    setTransaction,
+    minPrice,
+    showConfirmedAlert,
+    onSubmit,
+    password,
+    onPwdChange,
+    pwdError,
+    sending,
+    setSending,
+  }
+}
+
+export const useOutputs = ({
+  transaction,
+  isMainnet,
+  addresses,
+  sUDTAccounts,
+  fee,
+}: {
+  transaction: State.GeneratedTx | null
+  isMainnet: boolean
+  addresses: State.Address[]
+  sUDTAccounts: State.SUDTAccount[]
+  fee: bigint
+}) => {
+  const getLastOutputAddress = (outputs: State.DetailedOutput[]) => {
+    if (outputs.length === 1) {
+      return scriptToAddress(outputs[0].lock, isMainnet)
+    }
+
+    const change = outputs.find(output => {
+      const address = scriptToAddress(output.lock, isMainnet)
+      return !!addresses.find(item => item.address === address && item.type === 1)
+    })
+
+    if (change) {
+      return scriptToAddress(change.lock, isMainnet)
+    }
+
+    const receive = outputs.find(output => {
+      const address = scriptToAddress(output.lock, isMainnet)
+      return !!addresses.find(item => item.address === address && item.type === 0)
+    })
+    if (receive) {
+      return scriptToAddress(receive.lock, isMainnet)
+    }
+
+    const sudt = outputs.find(output => {
+      const address = scriptToAddress(output.lock, isMainnet)
+      return !!sUDTAccounts.find(item => item.address === address)
+    })
+    if (sudt) {
+      return scriptToAddress(sudt.lock, isMainnet)
+    }
+    return ''
+  }
+
+  const items: {
+    address: string
+    amount: string
+    capacity: string
+    isLastOutput: boolean
+    output: State.DetailedOutput
+  }[] = useMemo(() => {
+    if (transaction && transaction.outputs.length) {
+      const lastOutputAddress = getLastOutputAddress(transaction.outputs)
+      return transaction.outputs.map(output => {
+        const address = scriptToAddress(output.lock, isMainnet)
+        return {
+          capacity: output.capacity,
+          address,
+          output,
+          amount: shannonToCKBFormatter(output.capacity || '0'),
+          isLastOutput: address === lastOutputAddress,
+        }
+      })
+    }
+    return []
+  }, [transaction?.outputs])
+
+  const outputsCapacity = useMemo(() => {
+    const outputList = items.filter(item => !item.isLastOutput)
+    return outputList.reduce((total, cur) => {
+      return total + BigInt(cur.capacity || '0')
+    }, BigInt(0))
+  }, [items])
+
+  const lastOutputsCapacity = useMemo(() => {
+    if (transaction) {
+      const inputsCapacity = transaction.inputs.reduce((total, cur) => {
+        return total + BigInt(cur.capacity || '0')
+      }, BigInt(0))
+
+      return inputsCapacity - outputsCapacity - fee
+    }
+    return -1
```

Comment:
What does `-1` mean for `last output capacity`?

Reply:
Any update on this conversation @devchenyan

Reply:
When the transaction does not exist, it also returns a number that satisfies the type and can be used in the lastOutputsCapacity < 0 judgement.

Reply:
I guess it will be better to return `undefined` or `null` without a transaction. Then the disabled reason is no transaction rather than capacity is not enough.